zio-http-testkit/src/main/scala/zio/http/TestServer.scala

```diff
@@ -98,7 +98,7 @@ final case class TestServer(driver: Driver, bindPort: Int) extends Server {
       _ <- driver.addApp(provided, r)
     } yield ()
 
-  override def install[R](routes: Routes[R, Response])(implicit
+  override def install[R](routes: Routes[Scope & R, Response])(implicit
```

Comment:
What's the purpose of this `Scope` requirement?

I'd prefer hiding it from the `Routes` because it makes users to provide `Scope` that is not actually used.

Reply:
No, the routes is the right place. We need to close the Scope in the driver, after netty write is over.
If we eliminate the Scope earlier, we would need to close the scope earlier too.
Else you do route.run(..) and your scope is not closed/provided. That would be a bug

Reply:
By 'eliminating', I mean something like this:

```scala
trait Route[-Env, +Err] {
  def apply(request: Request): ZIO[Scope & Env, Response, Response] = ???
}

case class RoutePattern[A](...) {
  def ->[Env, Err, I](handler: Handler[Scope & Env, Err, I, Response]): Route[Env, Err] = ???
}
```

Unlike the other environments, `Scope` is a local requirement per request. I think this encoding reveals the relationship more clearly, but there's two downsides that we need to modify every route forming function including Endpoint DSL and the burden to the compiler of the added type calculation per route . In this respect, I think the current design in this PR is still good.

Reply:
@guersam I tried that out. It complicates the code, has negative impact on type inference and all just to save `Scope` on the type of `Routes` seems like a bad trade off

Reply:
> @guersam I tried that out. It complicates the code, has negative impact on type inference and all just to save `Scope` on the type of `Routes` seems like a bad trade off

This was my conclusion on https://github.com/zio/zio-http/pull/3229 as well, I agree with @987Nabil  this is the best approach to not make it very convoluted.